### PR TITLE
lightning,dumpling: make tidb version check more compitible

### PR DESCRIPTION
**Pull request by glorv**
### What problem does this PR solve?
TiDB version string can be customized by set the corresponding config field. So parse the result from `select verson()` to get tidb release version is not correct in some case.

### What is changed and how it works?
- Try to exec `select version()` and parse the result as tidb version; If the query returns error, fallback to `select version();` so we can still be compitible with MySQL/MariaDB. 

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
  - Run dumpling and lightning(local backend) with TiDB playground v5.2.2 (`server-version = 'Test-5.7.22'`), both can detect tidb version correctly and export/import successfully
  - Run dumpling and lightning (tidb backend) with MySQL 5.7.22, both can export/import successfully.
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
Avoid version check failure when tidb version is customized.
```

### CODE REVIEW

File: br/pkg/version/version.go [hunk 1]
```diff
@@ -255,32 +267,59 @@ func FetchVersion(ctx context.Context, db utils.QueryExecutor) (string, error) {
 type ServerType int
 
 const (
-	// ServerTypeUnknown represents unknown server type
+	// version.ServerTypeUnknown represents unknown server type
```
Comment (3pointer) [inline br/pkg/version/version.go:270]: 🤔

File: br/pkg/version/version.go [hunk 2]
```diff
@@ -241,11 +242,22 @@ func NormalizeBackupVersion(version string) *semver.Version {
 }
 
 // FetchVersion gets the version information from the database server
+//
+// NOTE: the executed query will be:
+// - `select tidb_version()` if target db is tidb
+// - `select version()` if target db is not tidb
 func FetchVersion(ctx context.Context, db utils.QueryExecutor) (string, error) {
 	var versionInfo string
+	const queryTiDB = "SELECT tidb_version();"
```
Comment (lichunzhu) [inline br/pkg/version/version.go:251]: When is this supported? Will we have a compatible issue?

Comment (kennytm) [reply br/pkg/version/version.go:251]: iirc we have the `tidb_version()` function since the very beginning...

Comment (lichunzhu) [reply br/pkg/version/version.go:251]: gotta

File: dumpling/export/util_test.go [hunk 3]
```diff
@@ -6,22 +6,23 @@ import (
...
 
 	data := [][]interface{}{
-		{ServerTypeUnknown, consistencyTypeNone, true},
-		{ServerTypeMySQL, consistencyTypeFlush, true},
-		{ServerTypeMariaDB, consistencyTypeLock, true},
-		{ServerTypeTiDB, consistencyTypeNone, true},
-		{ServerTypeTiDB, consistencyTypeSnapshot, false},
-		{ServerTypeTiDB, consistencyTypeLock, true},
+		{version.ServerTypeUnknown, consistencyTypeNone, true},
+		{version.ServerTypeUnknown, consistencyTypeFlush, true},
```
Comment (lichunzhu) [inline dumpling/export/util_test.go:18]:
```suggestion
		{version.ServerTypeMySQL, consistencyTypeFlush, true},
```

**Review by lichunzhu — APPROVED** [review]

**Review by 3pointer — APPROVED** [review]
LGTM

**Review by lichunzhu — APPROVED** [review]

**Pull request merged**